scripts/irods/test/test_quotas.py

```diff
@@ -97,3 +97,58 @@ def test_filter_out_groups_when_selecting_user__issue_3507(self):
+        pep_map = {
+            'irods_rule_engine_plugin-irods_rule_language': textwrap.dedent('''
+                acRescQuotaPolicy {
+                    msiSetRescQuotaPolicy("on");
+                }
+            '''),
+            'irods_rule_engine_plugin-python': textwrap.dedent('''
+                def acRescQuotaPolicy(rule_args, callback, rei):
+                    callback.msiSetRescQuotaPolicy('on')
+            ''')
+        }
+
+        try:
+            with temporary_core_file() as core:
+                core.add_rule(pep_map[self.plugin_name])
+                IrodsController().reload_configuration()
+
+                group_name = 'test_group_case_10_11_12'
+                file_name = 'test_file_case_10_11_12'
+                resc_name = 'ufs0_quota_case_10_11_12'
+                lib.make_file(file_name, 4096, contents='arbitrary')
+                lib.create_ufs_resource(self.admin, resc_name)
+
+                self.admin.assert_icommand(['iadmin', 'mkgroup', f'{group_name}_1'])
+                self.admin.assert_icommand(['iadmin', 'mkgroup', f'{group_name}_2'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_1', resc_name, '5000'])
+                self.admin.assert_icommand(['iadmin', 'sgq', f'{group_name}_2', resc_name, '10000'])
+
+                self.admin.assert_icommand(['iadmin', 'atg', f'{group_name}_1', self.user0.username])
+                self.admin.assert_icommand(['iadmin', 'atg', f'{group_name}_2', self.user0.username])
+
+                self.user0.assert_icommand(['iput', '-R', self.testresc, file_name])
+                self.admin.assert_icommand(['iadmin', 'cu'])
+
+                # Case 10: under quota succeeds
+                with self.subTest(testcase=10, desc="Under quota before and after"):
+                    self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_10'])
+                    self.admin.assert_icommand(['iadmin', 'cu'])
+
+                # Case 11: crossing quota succeeds
+                with self.subTest(testcase=11, desc="Crossing quota"):
+                    self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_11'])
+                    self.admin.assert_icommand(['iadmin', 'cu'])
+
+                # Case 12: should fail with just a single quota violated
+                with self.subTest(testcase=12, desc="Violating quota"):
+                    self.user0.assert_icommand(['icp', '-R', resc_name, file_name, f'{file_name}_case_12'], 'STDERR_SINGLELINE', 'SYS_RESC_QUOTA_EXCEEDED')
+        finally:
+            self.admin.run_icommand(['irm', '-f', f'{file_name}_case_10', f'{file_name}_case_11', f'{file_name}_case_12'])
+            self.admin.run_icommand(['iadmin', 'rmresc', resc_name])
+            self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_1'])
+            self.admin.run_icommand(['iadmin', 'rmgroup', f'{group_name}_2'])
```

Comment:
Missing a reload of the server configuration. Without it, the quota system will remain enabled.